Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -200,7 +233,45 @@ public struct TLSConfiguration {
 
     /// The pre-TLS1.3 cipher suites supported by this handler. This uses the OpenSSL cipher string format.
     /// TLS 1.3 cipher suites cannot be configured.
-    public var cipherSuites: String
+    public var cipherSuites: String = defaultCipherSuites
+    
+    /// Public property used to set the internal cipherSuiteFormattedString from NIOTLSCipher.
+    public var cipherSuite: [NIOTLSCipher] {
+        get {
+            // Set a temporary context with the default ciphers from cipherSuites
+            guard let tempContext = CNIOBoringSSL_SSL_CTX_new(CNIOBoringSSL_TLS_method()) else {
```

Comment:
Rather than use the unsafe API directly, can I propose constructing an `SSLContext` object that we can use? There are a number of ways the unsafe code can be misused (here we're leaking memory, for example), so I think it would be substantially better to extend the safe API on `SSLContext` to address our use-case.

In this case we can do this by simply passing `self` to the `SSLContext` (that is, fully initialising it) and then we can just add a getter that handles `SSL_CTX_get_ciphers` to get the stack.

We'll also want a safe abstraction to handle working with the stack. I've written most of one already in `PeerCertificateChainBuffers`: you could rewrite the majority of that data structure to provide a safe API for iterating `STACK_OF(SSL_CIPHER)`.

Reply:
> Rather than use the unsafe API directly, can I propose constructing an SSLContext object that we can use?

Yeah, I like that idea, especially to avoid the leak here.

Regarding:
> In this case we can do this by simply passing self to the SSLContext (that is, fully initialising it) and then we can just 
> add a getter that handles SSL_CTX_get_ciphers to get the stack.

How about something like the following:
```swift
public struct TLSConfiguration {
    ...

    public var cipherSuite: [NIOTLSCipher] {
        get {
            var defaultCiphers: [NIOTLSCipher] = []
            do {
                let sslConext = try NIOSSLContext(configuration: self)
                let cipherIDs = sslConext.getCiphersIDsFromContext()
                for id in cipherIDs {
                    defaultCiphers.append(NIOTLSCipher(id))
                }
            } catch {
                // Catch and return no ciphers
            }
            return defaultCiphers
        }
        set {
            ...
        }
    }
}


public final class NIOSSLContext {

   ...

   public func getCiphersIDsFromContext() -> [UInt16] {
        var cipherIDs: [UInt16] = []

        let bsslCipherStack = CNIOBoringSSL_SSL_CTX_get_ciphers(self.sslContext)
        /*
            Build abstraction for interating `STACK_OF(SSL_CIPHER)`
        */
        return cipherIDs
   }

}
```
Was this what you had in mind? (Minus the safe API for iterating the STACK_OF(SSL_CIPHER))

Reply:
I think we don't even need the ceremony of `getCipherIDsFromContext`: it can just be `var ciphers: [NIOTLSCipher]`.

Reply:
I refactored this code to adopt the safer approach here and also implemented the `NIOTLSCipherBuffers` struct in an attempt to safely navigate the `STACK_OF(SSL_CIPHER)`'s based on what you had for the `PeerCertificateChainBuffers`.  Would like to get your feedback here on the latest update.